openstack/glance/templates/etc/_glance-policy.json.tpl

```diff
@@ -1,50 +1,67 @@
 {
     "context_is_cloud_admin":  "role:cloud_image_admin",
     "context_is_admin":  "rule:context_is_cloud_admin",
+    "owner": "project_id:%(project_id)s",
+    "admin": "role:image_admin and rule:owner",
+    "member": "role:member and rule:owner",
+    "viewer": "role:image_viewer and rule:owner",
+    "context_is_image_admin":  "rule:context_is_admin or role:admin or rule:admin",
+    "context_is_image_viewer":  "rule:context_is_image_admin or rule:viewer or role:member",
     "default": "rule:context_is_admin",
-    "add_image": "",
-    "delete_image": "",
-    "get_image": "",
-    "get_images": "",
-    "modify_image": "",
-    "publicize_image": "rule:context_is_cloud_admin",
-    "copy_from": "",
-    "download_image": "",
-    "upload_image": "",
-    "delete_image_location": "",
-    "get_image_location": "",
-    "set_image_location": "",
-    "add_member": "",
-    "delete_member": "",
-    "get_member": "",
-    "get_members": "",
-    "modify_member": "",
+
+    "add_image": "rule:context_is_image_admin",
+    "delete_image": "rule:context_is_image_admin",
+    "get_image": "rule:context_is_image_viewer",
+    "get_images": "rule:context_is_image_admin",
+    "modify_image": "rule:context_is_image_admin",
+    "publicize_image": "rule:context_is_image_admin",
```

Comment:
only cloud admins should be able to do this, change back to `rule:context_is_cloud_admin` please